### DISCUSSION TITLE
Is there a way such that each tenant can have seperate S3 bucket with separately stored credentials?

### DISCUSSION THREAD

**Discussion by rnbokade** [neondatabase/neon#4139] [Q&A]
We are looking to selfhost this for a group of companies in our organization. Plan is to have one server with multiple compute processes for different companies through seperate tenants, dumping data of each to their own s3 instances.

Is it possible? If so then how?

Also is there any kubernetes helm chart or something which helps create new tenants with such parameters for each?

**Comment by martin-braun** [comment]
I'm also interested in this. We'd like to separate data per client, but have only one backend that connects to Neon, which will be spread across many servers behind a load balancer. If this is not possible, it would require us to separate Neon instances as well, which can be hell with many tenants.

The domain would be the given parameter that picks the right tenant and thus accesses data from the right S3 instance. I also wonder if it is possible have a cluster and each Neon instance will limit itself to a set of tenants, which would allow redundancy, but not overload the Neon instances. I need to do some more research in this topic, but it would be good to know if this is possible at all.

**Comment by muandane** [comment]
@rnbokade Hello i've been using minio for a while on AKS the normal version of minio is quite lacking in features, but if you use the Minio kubernetes operator you can create multiple Tenants and manage them with 1 operator.
minio operator helm chart -> minio tenant helm chart
[Minio Kubernetes Operator](https://github.com/minio/operator/tree/master/helm/operator)
